services/sn_analysis_service/app.py

```diff
@@ -0,0 +1,138 @@
+import functools
+import tempfile
+import io
+
+import numpy as np
+import matplotlib
+import arviz as az
+from quart import Quart
+from quart import request
+from astropy.table import Table
+import sncosmo
+
+from baselayer.log import make_log
+from baselayer.app.env import load_env
+
+_, cfg = load_env()
+log = make_log('sn_analysis_service')
+
+# we need to set the backend here to insure we
+# can render the plot headlessly
+matplotlib.use('Agg')
+rng = np.random.default_rng()
+
+app = Quart(__name__)
+app.debug = False
+
+default_analysis_parameters = {"source_name": "nugent-sn2p", "fix_z": False}
+
+
+async def run_sn_model(dd):
+    """
+    Use `sncosmo` to fit data to a model with name `source_name`.
+
+    We expect the `inputs` dictionary to have the following keys:
+       - source_name: the name of the model to fit to the data
+       - fix_z: whether to fix the redshift
+       - photometry: the photometry to fit to the model (in csv format)
+       - redshift: the known redshift of the object
+
+    TODO in the next PR: send the results back to the webhook
+
+    """
+    data_dict = await dd
+
+    analysis_parameters = data_dict["inputs"].get("analysis_parameters", {})
+    analysis_parameters = {**default_analysis_parameters, **analysis_parameters}
+
+    source_name = analysis_parameters.get("source_name")
+    fix_z = analysis_parameters.get("fix_z") in [True, "True", "t", "true"]
+
+    data = Table.read(data_dict["inputs"]["photometry"], format='ascii.csv')
```

Comment:
Maybe data checking on inputs at the top that photometry and redshift are there so it handles the error gracefully if not?